generators/docker-compose/index.js

```diff
@@ -117,7 +122,11 @@ module.exports = class extends BaseDockerGenerator {
                         yamlConfig.environment.push(`JHIPSTER_REGISTRY_PASSWORD=${this.adminPassword}`);
                     }
 
-                    parentConfiguration[`${lowercaseBaseName}-app`] = yamlConfig;
+                    if (!this.serviceDiscoveryType && appConfig.applicationType !== 'gateway') {
```

Comment:
This is the place which is misconfigured. We should do it only for microservices + uaa. Considering our future roadmap, a better condition would be `appConfig.skipClient`

Reply:
Thanks for clarifying @xetys. I made your requested change.